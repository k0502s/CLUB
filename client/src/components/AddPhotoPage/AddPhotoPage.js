import React, { useState, useEffect } from 'react';
import { Button, Form, Input, Label, Card, CardTitle, CardText, Row, Col } from 'reactstrap';
import FileUpload from '../utils/Fileupload';
import { useDispatch, useSelector } from 'react-redux';
import { PHOTO_UPLOADING_REQUEST } from '../../redux/types';

const Continents = [
    { key: 1, value: '풍경' },
    { key: 2, value: '인물' },
    { key: 3, value: '접사' },
    { key: 4, value: '습작' },
];

const AddPhotoPage = (props) => {
    const [Title, setTitle] = useState('');
    const [Description, setDescription] = useState('');
    const [Continent, setContinent] = useState(0);
    const [Images, setImages] = useState([]);
    const dispatch = useDispatch();

    const { user } = useSelector((state) => state.auth);

    const titleChangeHandler = (e) => {
        setTitle(e.currentTarget.value);
    };

    const descriptionChangeHandler = (e) => {
        setDescription(e.currentTarget.value);
    };

    const continentChangeHandler = (e) => {
        setContinent(e.currentTarget.value);
    };

    const updateImages = (newImages) => {
        setImages(newImages);
    };

    const submitHandler = (e) => {
        e.preventDefault();

        const title = document.myform.title.value;
        const description = document.myform.description.value;
        const continent = document.myform.continent.value;
        // if(name==="" || camera==="" || age==="" || sex===""){
        //     return alert("빈 칸에 정보를 입력해야 합니다.")
        // }
        if (title === '') {
            return alert('제목 정보를 입력해야 합니다.');
        }
        if (description === '') {
            return alert('설명 정보를 입력해야 합니다.');
        }
        if (continent === '') {
            return alert('장르 정보를 입력해야 합니다.');
        }
        if (Images === []) {
            return alert('사진을 입력해야 합니다.');
        }

        //서버에 채운 값들을 request을 보낸다.
        const body = {
            writer: user._id, //로그인된 사람의 ID
            title: Title,
            description: Description,
            images: Images,
            continents: Continent,
        };
        dispatch({
            type: PHOTO_UPLOADING_REQUEST,
            payload: body,
        });
    };

    return (
        <Card body style={{height:'700px', borderRadius:'50px'}}>
                <CardTitle tag="h3">
                    포토 추가하기
                </CardTitle>
                <Form onSubmit={submitHandler} name="myform">
                    {/* DropZone   */}
                    <FileUpload refreshFunction={updateImages} />
                    <Label>제목</Label>
                    <Input onChange={titleChangeHandler} value={Title} name="title" />
                    <br />
                    <Label>설명</Label>
                    <Input onChange={descriptionChangeHandler} value={Description} name="description" />
                    <br />
                    <Label>장르</Label>
                    <Input onChange={continentChangeHandler} value={Continent} name="continent" type="select">
                        <option value="">장르를 선택해주세요</option>
                        {Continents.map((item) => (
                            <option key={item.key} value={item.key}>
                                {item.value}
                            </option>
                        ))}
                    </Input>
                    <Button className="col-md-2 offset-md-10 mt-5" onClick={submitHandler}>
                        확인
                    </Button>
                </Form>
            </Card>
    );
};

export default AddPhotoPage;
