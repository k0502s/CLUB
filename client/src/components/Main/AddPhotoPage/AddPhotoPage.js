import React, { useState, useEffect } from 'react';
import { Helmet } from 'react-helmet';
import { Button, Form, Input, Label, Card, CardTitle, CardText, Row, Col, CardHeader, CardBody } from 'reactstrap';
import FileUpload from '../../../utils/Fileupload';
import { useDispatch, useSelector } from 'react-redux';
import { PHOTO_UPLOADING_REQUEST } from '../../../redux/types';

const Genres = [
    { key: 1, value: '풍경' },
    { key: 2, value: '인물' },
    { key: 3, value: '접사' },
    { key: 4, value: '습작' },
];

const AddPhotoPage = (props) => {
    const [form, setValues] = useState({
        title: '',
        description: '',
        genres: '',
    });
    const [Images, setImages] = useState([]);
    const dispatch = useDispatch();

    const { user } = useSelector((state) => state.auth);

    const onChange = (e) => {
        setValues({
            ...form,
            [e.target.name]: e.target.value,
        });
    };

    const updateImages = (newImages) => {
        setImages([...Images, newImages]);
    };

    const submitHandler = (e) => {
        e.preventDefault();

        const titleinput = document.myform.title.value;
        const descriptioninput = document.myform.description.value;
        const genresinput = document.myform.genres.value;

        if (titleinput === '') {
            return alert('제목 정보를 입력해야 합니다.');
        }
        if (descriptioninput === '') {
            return alert('설명 정보를 입력해야 합니다.');
        }
        if (genresinput === '') {
            return alert('장르 정보를 입력해야 합니다.');
        }
        if (Images.length === 0) {
            return alert('사진을 입력해야 합니다.');
        }

        const { title, description, genres } = form;

        const body = {
            writer: user._id,
            title: title,
            description: description,
            images: Images,
            genres: genres,
        };
        dispatch({
            type: PHOTO_UPLOADING_REQUEST,
            payload: body,
        });
    };

    const removefile = () => {
        setImages([]);
    };

    return (
        <Col md={{ size: 6, offset: 3 }} className="mt-4">
            <Helmet title={'포토 추가'} />
            <Card style={{ borderRadius: '25px', marginBottom: '20px' }}>
                <CardHeader>
                    <strong>포토 추가하기</strong>
                </CardHeader>
                <Card body>
                    <CardTitle tag="h5">※ 이미지 파일 업로드 참고 사항</CardTitle>
                    <CardText>1. 이미지 파일 사이즈는 100 * 1024 * 1024의 제한을 두고 있습니다.</CardText>
                    <CardText>2. 추가로 한 장 이상의 이미지 파일을 업로드 할 수 있습니다.</CardText>
                    <CardText>
                        <FileUpload refreshFunction={updateImages} removefile={removefile} />
                    </CardText>
                </Card>
                <CardBody>
                    <Col md={{ offset: 4 }} style={{ display: 'flex', width: '350px', height: '240px', overflow: 'scroll' }}>
                        {Images.map((image, index) => (
                            <img key={index} style={{ minWidth: '300px', width: '300px', height: '240px' }} src={`${image}`} />
                        ))}
                    </Col>
                    <Form onSubmit={submitHandler} name="myform">
                        <Label for="title">제목</Label>
                        <Input type="text" name="title" onChange={onChange} value={form.title} data-testid='add-title'/>
                        <br />
                        <Label for="content">설명</Label>
                        <Input type="textarea" name="description" onChange={onChange} value={form.description} data-testid='add-description'/>
                        <br />
                        <Label for="genres">장르</Label>
                        <Input type="select" name="genres" onChange={onChange} value={form.genre}>
                            <option value="">장르를 선택해주세요</option>
                            {Genres.map((item) => (
                                <option key={item.key} value={item.key} data-testid="select-option">
                                    {item.value}
                                </option>
                            ))}
                        </Input>
                        <Button className="col-md-2 offset-md-10 mt-5" onClick={submitHandler} data-testid='add-submit'>
                            확인
                        </Button>
                    </Form>
                </CardBody>
            </Card>
        </Col>
    );
};

export default AddPhotoPage;
