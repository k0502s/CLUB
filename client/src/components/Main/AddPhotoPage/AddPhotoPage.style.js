import styled from 'styled-components';
import { Col, Button, Card } from 'reactstrap';

const AddCard = styled(Card)`
    border-radius: 25px;
    margin-bottom: 0px;
    margin-top: 30px;
`;
const BtnWrap = styled.div`
    display: flex;
    justify-content: flex-end;
`;
const AddBtn = styled(Button)`
    background-color: ${(props) => props.color};
    margin: ${(props) => props.margin};
    width: ${(props) => props.width};
    color: #fff;
    &:hover {
        color: white;
        background-color: lightslategray;
    }
`;
const ProfileWarp = styled(Col)`
    display: flex;
    width: 350px;
    height: 240px;
    overflow: scroll;

    & img {
        min-width: 300px;
        width: 300px;
        height: 240px;
    }
`;

export { AddCard, ProfileWarp, AddBtn, BtnWrap };
