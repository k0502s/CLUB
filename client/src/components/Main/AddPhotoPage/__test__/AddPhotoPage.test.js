import React from 'react';
import AddPhotoPage from '../AddPhotoPage';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('<AddPhotoPage />', () => {
    const initialState = { output: 10 };
    const mockStore = configureStore();
    let store;

    it('matches snapshot', () => {
        store = mockStore(initialState);
        const utils = render(
            <Provider store={store}>
                <AddPhotoPage />
            </Provider>
        );
        expect(utils.container).toMatchSnapshot();
    });
});
