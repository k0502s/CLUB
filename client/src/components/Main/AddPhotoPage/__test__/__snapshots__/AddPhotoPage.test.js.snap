// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPhotoPage /> matches snapshot 1`] = `
<div>
  <div
    class="mt-4 col-md-6 offset-md-3"
  >
    <div
      class="card"
      style="border-radius: 25px; margin-bottom: 20px;"
    >
      <div
        class="card-header"
      >
        <strong>
          포토 추가하기
        </strong>
      </div>
      <div
        class="card card-body"
      >
        <h5
          class="card-title"
        >
          ※ 이미지 파일 업로드 참고 사항
        </h5>
        <p
          class="card-text"
        >
          1. 이미지 파일 사이즈는 100 * 1024 * 1024의 제한을 두고 있습니다.
        </p>
        <p
          class="card-text"
        >
          2. 추가로 한 장 이상의 이미지 파일을 업로드 할 수 있습니다.
        </p>
        <p
          class="card-text"
        >
          <section>
            <button
              class="mr-2 btn btn-secondary"
              tabindex="0"
              type="button"
            >
              <input
                autocomplete="off"
                multiple=""
                style="display: none;"
                tabindex="-1"
                type="file"
              />
              업로드
            </button>
            <button
              class="btn-danger btn btn-secondary"
              type="button"
            >
              파일 제거
            </button>
          </section>
        </p>
      </div>
      <div
        class="card-body"
      >
        <div
          class="offset-md-4"
          style="display: flex; width: 350px; height: 240px; overflow: scroll;"
        />
        <form
          class=""
          name="myform"
        >
          <label
            class=""
            for="title"
          >
            제목
          </label>
          <input
            class="form-control"
            name="title"
            type="text"
            value=""
          />
          <br />
          <label
            class=""
            for="content"
          >
            설명
          </label>
          <textarea
            class="form-control"
            name="description"
          />
          <br />
          <label
            class=""
            for="genres"
          >
            장르
          </label>
          <select
            class="form-control"
            name="genres"
          >
            <option
              value=""
            >
              장르를 선택해주세요
            </option>
            <option
              value="1"
            >
              풍경
            </option>
            <option
              value="2"
            >
              인물
            </option>
            <option
              value="3"
            >
              접사
            </option>
            <option
              value="4"
            >
              습작
            </option>
          </select>
          <button
            class="col-md-2 offset-md-10 mt-5 btn btn-secondary"
            type="button"
          >
            확인
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
